kernel/kernel-pdfgenerator/src/main/java/io/mosip/kernel/pdfgenerator/impl/PDFGeneratorImpl.java

```diff
@@ -0,0 +1,373 @@
+			builder.toStream(os);
+			builder.run();
+		} catch (Exception e) {
+			throw new PDFGeneratorException(PDFGeneratorExceptionCodeConstant.PDF_EXCEPTION.getErrorCode(),
+					e.getMessage(), e);
+		}
+		return os;
+	}
+
+	private String preprocessHtml(InputStream is) throws IOException {
+		// Specify UTF-8 encoding explicitly when reading the input stream
+		String html = new String(is.readAllBytes(), StandardCharsets.UTF_8);
+
+		// Parse the HTML content with Jsoup
+		Document document = Jsoup.parse(html);
+
+		// Preserve non-Latin characters by setting the correct output encoding
+		document.outputSettings()
+				.syntax(Document.OutputSettings.Syntax.xml)
+				.charset(StandardCharsets.UTF_8); // Ensure UTF-8 encoding for output
+
+		return document.html(); // Returns well-formed XHTML with proper encoding
+	}
+
+	@Override
+	public OutputStream generate(String template) throws IOException {
+		ByteArrayOutputStream os = new ByteArrayOutputStream();
+		if(template.isEmpty()){
+			throw new PDFGeneratorException(PDFGeneratorExceptionCodeConstant.INPUTSTREAM_NULL_EMPTY_EXCEPTION.getErrorCode(),
+					PDFGeneratorExceptionCodeConstant.INPUTSTREAM_NULL_EMPTY_EXCEPTION.getErrorMessage());
+		}
+		try {
+			Document document = Jsoup.parse(template);
+			document.outputSettings().syntax(Document.OutputSettings.Syntax.xml);
+			PdfRendererBuilder builder = FontPdfRendererBuilder.getBuilder();
+			builder.withHtmlContent(document.html(), null);
+			builder.toStream(os);
+			builder.run();
+		}catch (Exception e) {
+			throw new PDFGeneratorException(PDFGeneratorExceptionCodeConstant.PDF_EXCEPTION.getErrorCode(),
+					e.getMessage(), e);
+		}
+		return os;
+	}
+
+	@Override
+	public void generate(String templatePath, String outputFilePath, String outputFileName) throws IOException {
+		File outputFile = new File(outputFilePath + outputFileName + OUTPUT_FILE_EXTENSION);
+		try {
+
+
+		try (PDDocument document = new PDDocument()) {
+			PDPage page = new PDPage(PDRectangle.A4);
+			document.addPage(page);
+
+			try (BufferedReader reader = new BufferedReader(new FileReader(templatePath));
+				 PDPageContentStream contentStream = new PDPageContentStream(document, page)) {
+				contentStream.beginText();
+				contentStream.setFont(PDType1Font.HELVETICA, 12);
+				contentStream.newLineAtOffset(50, 750);
+
+				String line;
+				while ((line = reader.readLine()) != null) {
+					contentStream.showText(line);
+					contentStream.newLineAtOffset(0, -15); // Move down for the next line
+				}
+
+				contentStream.endText();
+			}
+
+			document.save(outputFile);
+		}
+		}catch (Exception e){
+			throw new PDFGeneratorException(PDFGeneratorExceptionCodeConstant.PDF_EXCEPTION.getErrorCode(),
+					e.getMessage(), e);
+		}
+	}
+
+	@Override
+	public OutputStream generate(InputStream is, String resourceLoc) throws IOException {
+		return generate(is); // Placeholder, as no specific resource handling is demonstrated in the existing implementation
```

Comment:
@dhanendra06  Unable to understand the comment here.

Reply:
removed this comments